lib/evm/interpreter.js

```diff
@@ -0,0 +1,206 @@
+const promisify = require('util.promisify')
+const BN = require('bn.js')
+const ethUtil = require('ethereumjs-util')
+const { ERROR } = require('../exceptions')
+const Loop = require('./loop')
+const { StorageReader } = require('../state')
+const PStateManager = require('../state/promisified')
+
+module.exports = class Interpreter {
```

Comment:
Would it make sense to make the Interpreter class just the index.js entry point of the whole module and therefore ease the API? Or is there a reason to keep this explicit?

Reply:
It's definitely possible if you'd prefer. I'm following the convention that `index` contains (almost) no source code, and only is there to expose code to other modules. Other modules would then only import what's exported in `index` and not individual source files.

Reply:
Ok, let's keep for now. Still thinking that at the end people will instantiate their own (eventually modified/extended) EVM or eWASMVM, which they will pass to the outer VM execution engine similar to the StateManager being passed now, and where they will likely also directly get the events from (so we need no binding to the outer VM instance any more). Is this a similar model you envisioned, or do you have got an alternative structure in mind?

Reply:
Yeah that's quite similar to what I was thinking. VM would bind all these different classes together (e.g. evm/ewasm interpreter, state manager, etc.). Users could instantiate VM possibly with their custom implementation of these classes, and call higher-level functions like `runTx` or `runBlock`, or they could use the classes themselves if they need only a specific aspect of the logic.